File: elixir/rna-transcription/rna_transcription.exs
defmodule RNATranscription do
  @nucleotides %{
    ?G => ?C,
    ?C => ?G,
    ?T => ?A,
    ?A => ?U
  }

  def to_rna(dna) do
    Enum.map(dna, &convert_nucleotide/1)
  end

  defp convert_nucleotide(nucleotide) do
    case Map.get(@nucleotides, nucleotide) do
      found -> found
      _ -> nucleotide
    end
  end
end

File: erlang/rna-transcription/src/rna_transcription.erl
-module(rna_transcription).

-export ([to_rna/1, test_version/0]).

to_rna(String) ->
  lists:map(fun transcribe_char/1, String).

transcribe_char(Char) ->
  case Char of
    $G -> $C;
    $C -> $G;
    $T -> $A;
    $A -> $U
  end.

test_version() ->
  1.
